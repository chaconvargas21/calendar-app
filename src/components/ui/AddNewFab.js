import React from "react";
import { useDispatch } from "react-redux";
import { uiOpenModal } from "../../actions/ui";

export const AddNewFab = () => {
    const dispatch = useDispatch();
  const handleOpenModal = () => {
        dispatch(uiOpenModal())
  };
  return (
    <button
      className="text-white text-2xl bg-blue-700 rounded-full bottom-6 right-6 fixed px-6 py-5"
      onClick={handleOpenModal}
    >
      <i className="fa-solid fa-plus"></i>
    </button>
  );
};
