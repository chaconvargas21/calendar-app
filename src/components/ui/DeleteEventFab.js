import React from "react";
import { useDispatch } from "react-redux";
import { eventStartDelete } from "../../actions/events";

export const DeleteEventFab = () => {
  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(eventStartDelete());
  };
  return (
    <button
      className="text-white text-2xl bg-red-700 rounded-full bottom-6 left-6 fixed px-6 py-5"
      onClick={handleDelete}
    >
      <i className="fa-solid fa-trash"></i>
    </button>
  );
};
