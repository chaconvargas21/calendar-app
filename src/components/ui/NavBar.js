import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startLogout } from "../../actions/auth";

export const NavBar = () => {
  const dispatch = useDispatch();
  const {name} = useSelector( state => state.auth );
  const handleLogout = () => {
    dispatch(startLogout());
  }
  return (
    <div className="flex justify-around content-center bg-gray-800 p-2">
      <div>
        <span className="text-white text-lg font-semibold">{name}</span>
      </div>
      <div>
        <button className="text-red-700 text-sm font-light border border-red-700 rounded hover:bg-red-700  hover:text-white p-2"
        onClick={handleLogout}>
          <i className="fa-solid fa-right-from-bracket"></i>
          <span className="ml-2">Salir</span>
        </button>
      </div>
    </div>
  );
};
