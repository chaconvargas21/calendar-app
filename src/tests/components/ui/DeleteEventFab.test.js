import { mount } from "enzyme"
import { Provider } from "react-redux"
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import { DeleteEventFab } from "../../../components/ui/DeleteEventFab";
import { eventStartDelete } from "../../../actions/events";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
jest.mock('../../../actions/events',()=>({
    eventStartDelete: jest.fn()
}))
const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn()
const wrapper = mount(
    <Provider store = {store}>
        <DeleteEventFab></DeleteEventFab>
    </Provider>
)

describe('Pruebas en DeleteEventFab', () => { 
    test('debe de mostrarse correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
     })
    test('debe de llamar eventStartDelete', () => { 
        wrapper.find('button').prop('onClick')();
        expect(eventStartDelete).toHaveBeenCalled();
     })
 })