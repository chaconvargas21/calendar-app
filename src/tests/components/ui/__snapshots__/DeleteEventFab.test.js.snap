// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DeleteEventFab debe de mostrarse correctamente 1`] = `
<button
  className="text-white text-2xl bg-red-700 rounded-full bottom-6 left-6 fixed px-6 py-5"
  onClick={[Function]}
>
  <i
    className="fa-solid fa-trash"
  />
</button>
`;
